### PR TITLE
#184, proper regexes for dates and times.

### PR DESCRIPTION
**Pull request by torbengb**
I like the exactness of this approach, but I admit that the long regex strings look daunting. That's why I also added the samples in the comment section. I'm aware that [you disagreed here](https://github.com/torbengb/bank2ynab/issues/182#issuecomment-368596883) but I'm testing this anyway... 

I feel it would be careless to specify the patterns only as precisely as necessary, because when every new format changes the definition of "as necessary". We're already seeing some conflicts, and leaving the formats vaguely referenced opens us up to mismatches when future formats arrive. I think there's value in being precise, especially as the number of supported formats increase.

### CODE REVIEW

File: bank2ynab.conf [hunk 1]
```diff
@@ -35,7 +35,19 @@ Plugin =
 # Just make a new [section heading] and copy only the lines from above that need a non-default value.
 # Please keep this sorted alphabetically by [country + bank name] for easier maintenance.
 #####################################################################################################
-# DATE FORMATS:
+# FILENAME REGEX DATE FORMATS:
+# YYYY       ([12][0-9]{3})
```
Comment (torbengb) [inline bank2ynab.conf]: Is there a way to turn these samples into actual building blocks that we could use in the `Source Filename Pattern`? 

I.e. could we turn `([12][0-9]{3})` into e.g. `%%m` and so on? If we could, we could shorten:
`(([12][0-9]{3})-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01]))` to just:
`(%%y-%%m-%%d)`. 
This is more elegant and more readable - but we're inventing a new syntax and that may be a bad idea?

Comment (nocalla) [reply bank2ynab.conf]: I think that's an unnecessary amount of work. We'd have to introduce a whole dictionary of terms. Stick with simple regex. Why are we using these crazy long regexes anyway?

**Comment by nocalla** [comment]
I don't see the point of this.

**Comment by torbengb** [comment]
> I think that's an unnecessary amount of work. We'd have to introduce a
whole dictionary of terms. Stick with simple regex. Why are we using these
crazy long regexes anyway?

I sense that I am upsetting you with this, aren't I? That's not my
intention!

I think there's value in being precise, especially as the number of
supported formats increase. I feel it would be careless to specify the
patterns only as precisely as necessary, because when every new format
changes the definition of "as necessary". We're already seeing some
conflicts, and leaving the formats vaguely referenced opens us up to
mismatches when future formats arrive.

My goal was to prevent mismatches between dates and other "random" numbers.
If we specify that "this is an 8-digit date" then it won't match any other
8-digit numbers - or at least, the risk becomes so much lower.

**Comment by torbengb** [comment]
I'll withdraw this for now because it sticks out like a sore thumb. I'll keep the branch around until we decide exactly how to handle conflicts (also that bug) - perhaps parts of this might be useful, the rest can go.

**Pull request closed without merging**